### ISSUE TITLE
I am on Ubuntu 22 with amd 7900 xtx GPU... during install process it tries to install pip torch rocm dependancies from https://download.pytorch.org/whl/nightly/rocm5.6 but 5.6 is deprecated so it fails, the correct version is rocm6.0. Can you update it to install from https://download.pytorch.org/whl/nightly/rocm6.0 please?

### ISSUE THREAD

**Issue by ciaranofseir** [easydiffusion/easydiffusion#1745]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: 
 - Browser:
 - Version:

**Smartphone (please complete the following information):**
 - Device: 
 - OS: 
 - Browser 
 - Version 

**Additional context**

**Comment by CataCluj** [comment]
Indeed, it needs to work with latest ROCm

**Comment by brandonjank** [comment]
same issue

**Comment by cmdr2** [comment]
@brandonjank @CataCluj @ciaranofseir Sorry for the extremely late reply, I was on a bit of a break from this project. If this is still relevant, I've updated the list of supported ROCm versions.

Can you please try this:

1. In a terminal, run `./developer_console.sh` inside the Easy Diffusion folder
2. Then run: `python -m pip install --upgrade torch torchvision --index-url https://download.pytorch.org/whl/nightly/rocm6.2`
3. Then open a new terminal tab, and run Easy Diffusion.

Does it generate the image? Please paste the output of the command prompt window after you run Easy Diffusion, I'm curious to know how it handles the new version of torch. Thanks!

**Comment by CataCluj** [comment]
I'm a noob at this, so I may be looking at the wrong things... but latest ROCm seems to be 6.3.1.
The instructions above specify 6.2 and the readme still says 5.2.
Is there no way in Linux to just make things work with "latest" or have multiple versions installed in the system (not containers) and have "programs" use what they need from what's available?

**Comment by cmdr2** [comment]
Hi @CataCluj I've updated the readme to say "rocm 5.2 (or newer)", since that's just specifying the minimum system requirements.

I've also updated my instructions in the previous comment, and that'll install the latest stable combination of torch+rocm (as per the PyTorch team).

> latest ROCm seems to be 6.3.1.
The ROCm version considered stable by torch might be a bit older than the latest release of ROCm, since the torch team needs some time to test and release newer versions of torch for that.

> Is there no way in Linux to just make things work with "latest" or have multiple versions installed in the system (not containers) and have "programs" use what they need from what's available?

I totally understand what you mean, but if you install system-wide packages, there can be only one version active at a time. Plenty of people use multiple AI projects (image generators, text generators etc), so it's easy to break each other. So that's why ED just runs in a lightweight conda environment. It's not as heavy as a container, but slightly heavier than python `venv`.

Python `venv` or `conda` is basically designed to address what you're asking, without using containers.

**Comment by CataCluj** [comment]
Thank you @cmdr2,
That makes much better! Also, thank you for the kind explanations 🙂
You can tell I'm a noob, and maybe that's why I have further ... wishes(?):

-  I wish the latest "stable" was installed when one specifies "latest" instead of having to specify an older/stable version. There should be another specifier like "beta" or whatever if one wants the bleeding edge.
-  Venv or conda may be lighter, but still create dupli/multiplication of common components which can be quite large. Sure storage may be cheap, but managing it can be time consuming, at least for people like me. Deleting a venv, because I don't know what's wrong with it just to re-create it with some changes, again and again... especially since they can be very complex; trying things different people post in different forums...
-  Separately, being a noob, I always worry about using something running in one venv from another one. And (being a noob) I find it difficult to combine two venvs into one, not to mention they may use different versions of... things. You get a venv/container for STT, another for LLM, another for TTS, all with their different requirements.
-  I may not understand the "active version" well enough. I thought it's just about paths in system path env var or something like that. Maybe wrongly, I think that, since multiple versions of python/others can be installed in the system, other mechanisms to select the wanted versions should exist. I think (but being a noob I'm not sure) other (hated?) OSes manage to do that. I'm not suggesting those are "better" (even if they are for some) but why can't a progressive (?) OS take what's best from everywhere?

Sorry for maybe throwing the wrong things in there; I don't mean to offend anyone.

________________________________

**Comment by cmdr2** [comment]
@CataCluj In terms of storage, both approaches would take the same amount of storage - i.e. whether each container/venv stores a separate copy of the package, or whether there's a system-wide collection of packages that stores different versions (and each software picks the version that it needs). [a], [b], [c] vs [a, b, c] -> both take the same amount of storage.

re: "stable" vs "beta" - Not sure if you were referring to Easy Diffusion, but yeah that's what Easy Diffusion does. Everyone uses the "stable" branch, which contains a well-tested latest version. And people can choose to use the "beta" branch (from the Settings tab), and then they'll get the bleeding edge version of Easy Diffusion. Easy Diffusion auto-updates each time it starts, so you're always running on the latest version in your selected branch.

**Comment by CataCluj** [comment]
Thank you @cmdr2
Is that still true if a, b, c all need d? and e, f, g need h? and a, b, e need i?

re: "stable" vs "beta" I was talking in general. So then instead of having to specify 6.2 I should be able to say "latest"

I think many installation scripts or requirements specify a version that in time becomes old, and it feels impossible to me (or just impractically hard) to find out if changing that to a newer/current version might work. If one could just specify a major/stable and if that was already installed it would save a lot, or if new, would be installed in system and might be reused by other programs. I think (but I'm not sure) like .NET versions.

**Comment by cmdr2** [comment]
> If one could just specify a major/stable and if that was already installed it would save a lot, or if new, would be installed in system and might be reused by other programs

Yes, that's what happens, but in practice, dependencies sometimes introduce breaking bugs, and over time, programs like Easy Diffusion end up with a bunch of specific versions for dependencies to work-around issues. Python's package manager (pip) works with dependency specifications like ">=2.0", "<2.4" etc.
